src/vision/status_quo/aws_engineer/missed_waker_leads_to_lost_performance.md

```diff
@@ -0,0 +1,12 @@
+# Status quo of an AWS engineer: Missed `Waker` leads to lost performance
+
+Once the server is working, Alan starts to benchmark it. He is not really sure what to expect, but he is hoping to see an improvement in performance relative to the baseline service they were using before. To his surprise, it seems to be running slower!
+
+Digging into the problem a bit, he wishes -- not for the first time -- that he had better tools to understand what was happening. This time he doesn't even bother trying to fire up the debugger. He just starts looking at his metrics. He's accumulated a pretty decent set of metrics by now, and he can often get a picture of the state of the runtime from them.
+
+After a few days of poking at the problem, Alan notices something odd. It seems like there is often a fairly large delay between the completion of a particular event and the execution of the code that is meant to respond to that event. Looking more closely, he realizes that the code for handling that event fails to trigger the `Waker` associated with the future, and hence the future never wakes up.
```

Comment:
It would be interesting to read how Alan's poking eventually lead him to the root cause.  Was his set of metrics rich enough to discover this without using any other debugging techniques?  Or did he need to do something like println debugging with timestamps?